qiita_db/data.py

```diff
@@ -416,6 +416,17 @@ def _remove_filepath(self, fp, conn_handler, queue):
             self._set_link_filepaths_status("failed: %s" % msg)
             raise QiitaDBError(msg)
 
+        # The filepath belongs to one or more studies
+        studies_linked = self.studies
```

Comment:
I think is fine to have it here but I see your point about adding more files. IMOO the best behavior will be to not allow linking more file paths if some have been already added. If you really want to add more you will have to unlink and then link again.